src/api/export.tsx

```diff
@@ -62,17 +63,42 @@ export const exportData = async (
   const existsConditions: { [x: string]: { $exists: boolean } }[] = [];
 
   projection?.forEach((channel) => {
+    // Do not project on functions
+    let is_function = false;
+    functionsState.functions.forEach((func) => {
+      if (channel === func.name) is_function = true;
+    });
+
     // API recognises projection values as metadata.key or channel.key
     // Therefore, we must construct the appropriate parameter
     const key =
       channel in staticChannels ? `metadata.${channel}` : `channels.${channel}`;
     queryParams.append('projection', key);
 
-    if (!(channel in staticChannels)) {
+    if (!(channel in staticChannels) && !is_function) {
```

Comment:
Just a minor comment unrelated to your change - we need to change the `!(channel in staticChannels)` to `channel !== timeChannelName` - as e.g. shot number and active experiment can both be undefined (it's probably worth going through the EPAC data interchange format document and checking our typing is all correct as well at some point)

I think this issue is also in the records and plot records endpoint as well